File: go/internal/handlers/health.go
package handlers

import (
	"net/http"

	"github.com/labstack/echo/v4"
	"github.com/microservices-boilerplate/go/internal/config"
)

// HealthHandler provides health check endpoints.
type HealthHandler struct {
	cfg *config.Config
}

// NewHealthHandler creates a new HealthHandler.
func NewHealthHandler(cfg *config.Config) *HealthHandler {
	return &HealthHandler{cfg: cfg}
}

// Liveness returns a simple alive check.
func (h *HealthHandler) Liveness(c echo.Context) error {
	return c.JSON(http.StatusOK, map[string]string{
		"status":  "alive",
		"service": h.cfg.AppName,
	})
}

// Readiness returns a readiness check (extend for DB, etc.).
func (h *HealthHandler) Readiness(c echo.Context) error {
	return c.JSON(http.StatusOK, map[string]string{
		"status":  "ready",
		"service": h.cfg.AppName,
	})
}

File: python/app/routers/health.py
"""Health check endpoints for readiness and liveness probes."""

from fastapi import APIRouter

from app.config import get_settings

router = APIRouter(tags=["Health"])
settings = get_settings()


@router.get("/health", summary="Liveness probe")
async def liveness() -> dict:
    """
    Simple liveness check - returns 200 if the service is running.
    Use for Kubernetes/Docker health checks.
    """
    return {"status": "alive", "service": settings.app_name}


@router.get("/health/ready", summary="Readiness probe")
async def readiness() -> dict:
    """
    Readiness check - verifies the service can accept traffic.
    Extend to check DB connectivity, external services, etc.
    """
    return {"status": "ready", "service": settings.app_name}
